sdk/search/azure-search-documents/azure/search/documents/agent/_agent_client.py

```diff
@@ -0,0 +1,94 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+from typing import Any, Union, Optional, IO
+
+from azure.core.credentials import AzureKeyCredential, TokenCredential
+from azure.core.tracing.decorator import distributed_trace
+from .._api_versions import DEFAULT_VERSION
+from ._generated import KnowledgeAgentRetrievalClient as _KnowledgeAgentRetrievalClient
+from ._generated.models import (
+    KnowledgeAgentRetrievalRequest,
+    RequestOptions,
+    KnowledgeAgentRetrievalResponse,
+)
+from .._headers_mixin import HeadersMixin
+from .._utils import get_authentication_policy
+from .._version import SDK_MONIKER
+
+
+class KnowledgeAgentRetrievalClient(HeadersMixin):
+    """A client that can be used to query an agent.
+
+    :param endpoint: The URL endpoint of an Azure search service
+    :type endpoint: str
+    :param agent_name: The name of the agent. Required.
+    :type agent_name: str
+    :param credential: A credential to authorize search client requests
+    :type credential: ~azure.core.credentials.AzureKeyCredential or ~azure.core.credentials.TokenCredential
+    :keyword str api_version: The Search API version to use for requests.
+    :keyword str audience: sets the Audience to use for authentication with Azure Active Directory (AAD). The
```

Comment:
Nit: Should go through all the clients where `audience` is a kwarg and rename `Azure Active Directory (AAD)` to `Microsoft Entra ID`.